service-loadbalancer/service_loadbalancer.go

```diff
@@ -322,9 +338,19 @@ func (lbc *loadBalancerController) getServices() (httpSvc []service, tcpSvc []se
 				Name: getServiceNameForLBRule(&s, servicePort.Port),
 				Ep:   ep,
 			}
+
 			if val, ok := s.Labels[lbHostKey]; ok {
```

Comment:
I want to encapsulate all the annotations logic so it's clear how someone would add support for a new annotation by looking at the code. 
Suggest something like: 
```
type serviceAnnotations map[string]string

func (s serviceAnnotations)getAlgorithm() {
    return s[lbAlgorightmsKey]
}
...
s.Labels.(serviceAnnotations).getAlgorightm()
```

Also why are we using labels instead of annotations (I'm not against it, just asking)? See [apiconvention](https://github.com/kubernetes/kubernetes/blob/11c31aeca7ec8931150ba8f6ceab86c38155fb9c/docs/devel/api-conventions.md#label-selector-and-annotation-conventions ) on this
if you still feel labels are the right thing, then this is ok, but elaborate on your reasoning in a comment.

```
Annotations have very different intended usage from labels. We expect them to 
be primarily generated and consumed by tooling and system extensions. I'm inclined
to generalize annotations to permit them to directly store arbitrary json. Rigid names 
and name prefixes make sense, since they are analogous to API fields.
```

If you want to keep them in labels because they're easier to play around with for now, but plan to move them to annotation soon, then please add a TODO.